src/plans.js

```diff
@@ -635,7 +635,7 @@ function insertJobs (schema) {
 function purge (schema, interval) {
   return `
     DELETE FROM ${schema}.archive
-    WHERE archivedOn < (now() - interval '${interval}')
+    WHERE archivedOn < (now() - interval '${interval}') RETURNING id
```

Comment:
Let's make this conditional (opt in) via config like state monitoring so it's not producing any data by default

Reply:
Ok. I will have a look how maintance is implemented. 
What do you think about some few more events?
Like canceled, resumed, failed, created, archived, completed, started and so on?
So if someone has a monitoring ui it is possible to send events via socket or so?

Reply:
Yes, `failed` could potentially be an event, but you could also use onComplete(), which is distribution friendly (see comment below about load balancing). Cancelled would only occur after calling cancel(), so the caller would already have the job ids. For creation, it's more complicated since batch insertion or direct table insert is supported that would bypass send() that returns the job id. The `wip` event will give you somewhat of an idea of when jobs are starting, but it doesn't emit per job. 

Another thing to consider when using events is that they only track activity from a single instance, so if you're load balancing with additional instances, you won't get events from others.

Reply:
Another thing to consider when using events is that they only track activity from a single instance, so if you're load balancing with additional instances, you won't get events from others.
You are completly right. It might not be useful for many. I run pg-boss in an Micro-service enviroment, so each instance can send its events via a central eventbus. 
i already send this events in work, and workCompleted, but you have to do it in every worker.
I will modify purge like you said and implement all other events in custom code or write a wrapper class